composites/Plugin/SnippetEditor/components/SnippetEditorFields.js

```diff
@@ -205,36 +205,45 @@ class SnippetEditorFields extends React.Component {
 	 */
 	render() {
 		const {
-			replacementVariables,
-			onChange,
-			onFocus,
-			data,
 			activeField,
 			hoveredField,
+			replacementVariables,
 			titleLengthProgress,
 			descriptionLengthProgress,
+			onFocus,
+			onChange,
+			data: {
+				title,
+				slug,
+				description,
+			},
 		} = this.props;
 
-		const { title, slug, description } = data;
+		const titleLabelId = `${ this.uniqueId }-title`;
+		const slugLabelId = `${ this.uniqueId }-slug`;
+		const descriptionLabelId = `${ this.uniqueId }-description`;
 
 		return (
 			<StyledEditor>
 				<FormSection>
 					<SimulatedLabel
-						id={ this.uniqueId + "-title" }
-						onClick={ () => onFocus( "title" ) }
-					>{ __( "SEO title", "yoast-components" ) }</SimulatedLabel>
-					<InputContainer isActive={ activeField === "title" } isHovered={ hoveredField === "title" }>
+						id={ titleLabelId }
+						onClick={ () => onFocus( "title" ) } >
+						{ __( "SEO title", "yoast-components" ) }
+					</SimulatedLabel>
+					<InputContainer
+						onClick={ () => this.elements.title.focus() }
+						isActive={ activeField === "title" }
+						isHovered={ hoveredField === "title" }>
 						<ReplacementVariableEditor
 							content={ title }
 							onChange={ content => onChange( "title", content ) }
 							onFocus={ () => onFocus( "title" ) }
 							replacementVariables={ replacementVariables }
 							ref={ ( ref ) => this.setRef( "title", ref ) }
-							ariaLabelledBy={ this.uniqueId + "-title" }
+							ariaLabelledBy={ titleLabelId }
 						/>
```

Comment:
Not sure if there's a specific reason for this change, it will result in a value that's just a number pointing to the ID above at line 230 that's just a number, see screenshot below:

<img width="878" alt="screen shot 2018-06-18 at 10 21 08" src="https://user-images.githubusercontent.com/1682452/41526751-6e56e888-72e5-11e8-97f9-dc820f65c58d.png">

I guess it would be safer to have a more qualified ID/aria-labelledby value to avoid potential conflicts with other IDs (you never know). This should apply also to the description field. As far as I see, the slug field still uses a more qualified value `this.uniqueId + "-slug" `. Will open a separate issue.